src/dom/progress-watcher.ts

```diff
@@ -0,0 +1,165 @@
+
+import { is } from '../is/is';
+import { mathf } from '../mathf/mathf';
+
+export interface ProgressWatcherItem {
+    range: number | Array<number>;
+    callback: Function;
+    inactiveCallback?: Function;
+    runWhen?: Function;
+}
+
+
```

Comment:
Nit: There seems to be a lot of extraneous newlines in this file, and I noticed it's 4-space indented and I believe other files are using 2-spaces. Is there an automatic linter/fixer in this proj?

Reply:
I'll look into integrating google style rules for the project.